src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseCountProperly.cs

```diff
@@ -0,0 +1,620 @@
+            else if (!isAsync)
+            {
+                AnalyzeCountInvocationOperation(context, invocationOperation);
+            }
+        }
+
+        private static void AnalyzePropertyReference(OperationAnalysisContext context)
+        {
+            var propertyReferenceOperation = (IPropertyReferenceOperation)context.Operation;
+            IOperation parentOperation = propertyReferenceOperation.Parent;
+
+            parentOperation = parentOperation.WalkUpParentheses();
+            parentOperation = parentOperation.WalkUpConversion();
+
+            // Analyze binary operation.
+            if (parentOperation is IBinaryOperation parentBinaryOperation)
+            {
+                AnalyzeParentBinaryOperation(context, propertyReferenceOperation, parentBinaryOperation,
+                    isAsync: false, methodName: null);
+            }
+            // Analize invocation operation, potentially obj.Count.Equals(0).
+            else if (parentOperation is IInvocationOperation parentInvocationOperation)
+            {
+                AnalyzeParentInvocationOperation(context, propertyReferenceOperation, parentInvocationOperation,
+                    isInstance: true, isAsync: false, methodName: null);
+            }
+            // Analize argument operation, potentially 0.Equals(obj.Count).
+            else if (parentOperation is IArgumentOperation argumentOperation)
+            {
+                AnalyzeParentInvocationOperation(context, propertyReferenceOperation, (IInvocationOperation)argumentOperation.Parent,
+                    isInstance: false, isAsync: false, methodName: null);
+            }
+        }
+
+        private static void AnalyzeParentBinaryOperation(
+            OperationAnalysisContext context, IOperation operation, IBinaryOperation parent, bool isAsync, string? methodName)
+        {
+            bool useRightSideExpression = default;
+            if (!IsLeftCountComparison(parent, out bool shouldNegateIsEmpty))
+            {
+                if (!IsRightCountComparison(parent, out shouldNegateIsEmpty))
+                {
+                    return;
+                }
+
+                useRightSideExpression = true;
+            }
+
+            if (isAsync)
+            {
+                bool shouldNegateAny = !shouldNegateIsEmpty;
+                ReportCA1828(context, shouldNegateAny, useRightSideExpression ? OperationBinaryRight : OperationBinaryLeft, methodName!, parent);
+            }
+            else
+            {
+                ITypeSymbol type = operation.GetInstanceType();
+
+                if (TypeContainsVsibileProperty(context, type, IsEmpty, SpecialType.System_Boolean))
+                {
+                    ReportCA1836(context, useRightSideExpression, shouldNegateIsEmpty, parent);
+                }
+                else
```

Comment:
I think we should change this to:
```suggestion
                else if (methodName != null)
```

Reply:
That's a good idea but now we will be mixing the purpose of the parameter, I can add a new parameter `bool allowLinq` to effectively avoid reporting Any() when the user was not even using Linq.